src/Parser/Core/Modules/Items/DeceiversGrandDesign.js

```diff
@@ -0,0 +1,126 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+
+import Module from 'Parser/Core/Module';
+
+const debug = false;
+const GUIDING_HAND_DURATION = 120000;
+const PROC_EVENT_START_BUFFER = 5000;
+const PROC_EVENT_END_BUFFER = 1000;
+
+class DecieversGrandDesign extends Module {
+  healing = 0;
+  healingAbsorb = 0;
+  dgdTargetOne = null;
+  gdgDownTimeTargetOne = 0;
+  dgdProcTimestampTargetOne = null;
+  gdgDownTimeTargetTwo = 0;
+  dgdTargetTwo = null;
+  dgdProcTimestampTargetTwo = null;
+  dgdProced = false;
+
+  dgdProcs = [];
+
+  on_initialized() {
+    if (!this.owner.error) {
+      this.active = this.owner.selectedCombatant.hasTrinket(ITEMS.DECEIVERS_GRAND_DESIGN.id);
+    }
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+
+    if(spellId === SPELLS.GUIDING_HAND.id) {
+      this.healing += event.amount || 0;
+      this.healing += event.absorbed || 0;
+
+      // Account for precasting
+      if(this.dgdTargetOne === event.targetID || this.dgdTargetTwo === event.targetID) {
+        return;
+      } else {
+        if(!this.dgdTargetOne) {
+          this.dgdTargetOne = event.targetID;
+        } else {
+          this.dgdTargetTwo = event.targetID;
+        }
+      }
+    }
+  }
+
+  on_byPlayer_absorbed(event) {
+    const spellId = event.ability.guid;
+
+    if(spellId === SPELLS.FRUITFUL_MACHINATIONS.id) {
+      this.healingAbsorb += event.amount || 0;
+      this.healingAbsorb += event.absorbed || 0;
```

Comment:
nit: I think this could be cleaner in one line;

this.healingAbsorb += (event.amount || 0) + (event.absorbed || 0);